app/models/generic_object_definition.rb

```diff
@@ -49,6 +45,31 @@ def property_defined?(attr)
     property_attribute_defined?(attr) || property_association_defined?(attr) || property_method_defined?(attr)
   end
 
+  def create_object(options)
+    GenericObject.create!({:generic_object_definition => self}.merge(options))
+  end
+
+  # To query based on GenericObject AR attributes and property attributes
+  #   find_objects(:name => "TestLoadBalancer", :uid => '0001', :prop_attr_1 => 10, :prop_attr_2 => true)
+  #
+  # To query based on property associations
```

Comment:
`To query based on property associations. The array does not need to be a complete list, but must contain only AR ids.`